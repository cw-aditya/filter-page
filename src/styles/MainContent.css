.main {
    display: flex;
    flex-direction: column;
}

.top-filter {
    width: 100%;
    display: flex;
    justify-content: end;
    background-color: #fff;
    border-radius: 8px;
}

.sort-by {
    display: flex;
    /* height: 40px; */
}

.sort-label{
    margin: 0;
    margin-right: 8px;
    margin-top: auto;
    margin-bottom: auto;
}

.sort-by-select {
    height: 36px;
    padding: 8px 37px 8px 12px;
    margin-bottom: 8px;

    margin-top: 8px;
}

.main-content {
    width: 745px;
    height: 100%;
    /* background-color: aqua; */
    padding: 8px;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(255px, 1fr));
}

.card {
    /* display: flex;
    flex-direction: column; */
    /* padding: 8px; */
    border-radius: 16px;
    background-color: white;
    margin: 8px;
    /* justify-content: start; */
}

.car-data {
    display: flex;
    flex-direction: column;
    padding: 12px;
    padding-bottom: 16px;
}


.image-wrapper {
    width: 100%;
    height: 0%;
    position: relative;
    padding: 0px;
    padding-bottom: 75%;
    /* box-sizing: border-box; */
    background-color: #EFEFEF;
}

.no-image {
    max-height: 112px;
    max-width: 174px;
    height: 100% !important;
    width: 100% !important;
    margin-top: auto;
    margin-bottom: auto;
    justify-self: center;
    /* position: absolute; */
    /* left: 50% !important;
    top: 50% !important;
    transform: translate(-50%, -50%); */
}

.car-image {
    width: 100%;
    height: auto;
    margin: 0px;
    left: 0px;
    right: 0px;
    bottom: 0px;
    top: 0px;
    /* position: absolute; */
    box-sizing: border-box;
}


.car-name {
    font-size: 15px;
    font-weight: 600;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;

}

.car-details {
    font-size: 13px;
    font-weight: 400;
}

.car-pricing {
    display: flex;
    justify-content: space-between;
    margin-top: 8px;
    margin-bottom: 16px;
}

.car-price {
    font-size: 18px;
    font-weight: 700;
    margin: 0px;
}

.car-emi {
    font-size: 16px;
    font-weight: 700;
}

.car-get-seller-details-btn {
    background-color: #E03012;
    border: none;
    height: 40px;
    border-radius: 6px;
}

.car-get-seller-details-btn>span {
    color: white;
    font-size: 16px;
    font-weight: 500;
}