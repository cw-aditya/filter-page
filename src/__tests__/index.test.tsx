import { describe, expect, test } from 'vitest'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
// import userEvent from '@testing-library/user-event';
import BudgetFilter from '@/components/BudgetFilter.component'
import FuelFilter from '@/components/FuelFilter.component';
import MainContent from '@/components/mainContent.component';
import Card from '@/components/Card.component';
import { log } from 'console';

// describe('A truthy statement', () => {
//     it('renders the App component', () => {
//         render(<App/>)

//         // screen.debug(); // prints out the jsx in the App component unto the command line
//       })
// })



describe("Budget Filter Component Test", () => {
  const setup = () => {
    return render(
      <BudgetFilter />
    )
  }

  test("Should Contain 2 input numbers", () => {
    setup()
    const inputs = screen.getAllByRole('spinbutton')
    expect(inputs).toHaveLength(2)
  })
  test("Should Render 2 input numbers", () => {
    setup()
    const inputs = screen.getAllByRole('spinbutton')
    inputs.forEach(element => {
      expect(element).toBeInTheDocument()

    });
  })
  test("Budget Component Snapshot", () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });
})

describe('Fuel Filter Component', () => {
  const setup = () => {
    return render(
      <FuelFilter />
    )
  }
  test('Should Contain 6 Checkboxes', () => {
    setup()
    const inputs = screen.getAllByRole('checkbox')
    expect(inputs).toHaveLength(6)
  });
  test('Should Render 6 Checkboxes', () => {
    setup()
    const inputs = screen.getAllByRole('checkbox')
    inputs.forEach(element => {
      expect(element).toBeInTheDocument()
    });
  });
  // test('Should Update when input changes', async () => {
  //   setup()
  //   const input = screen.getAllByRole('checkbox')[0]
  //   await userEvent.click(input)
  //   expect(input).toBeChecked()
  // });
  test("Fuel Filter Component Snapshot", () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });
})


describe('Main Component Tests', () => {

  const setup = () => {
    return render(<MainContent />)
  }

  test('Should render sort by filter', () => {
    setup()
    const sortByFilter = screen.getByRole('combobox')
    expect(sortByFilter).toBeInTheDocument()
  });
  test('Should have 3 filter options', () => {
    setup()
    const sortByFilterOptions = screen.getAllByRole('option')
    expect(sortByFilterOptions).toHaveLength(3)
  });
  test('Should render sort by filter options', () => {
    setup()
    const sortByFilterOptions = screen.getAllByRole('option')
    sortByFilterOptions.forEach(element => {
      expect(element).toBeInTheDocument()
    });
  });
  test("Main Component Snapshot", () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });
});

describe('Car Card Component Test', () => {
  const setup = () => {
    const props = {
      imageUrl: 'https://imgd.aeplcdn.com/0x0/cw/static/icons/svg/usedcar/no-car-image-with-text.svg',
      carName: 'Some Car Name',
      km: 50000,
      fuel: 'Petrol',
      cityName: 'Mumbai',
      price: '5 Lakh',
      emiText: 'EMI Starts at â‚¹4,751'
    }
    return render(<Card car={props} />)
  }
  test('Should render car name & price', () => {
    setup()
    const heading = screen.getAllByRole('heading')
    heading.forEach(element => {
      expect(element).toBeInTheDocument()
    });
  });
  test('Should render car image', () => {
    setup()
    const image = screen.getByAltText('Car image not available')
    // console.log(image);

    expect(image).toBeInTheDocument()
  });
  test("Card Component Snapshot", () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });

});

