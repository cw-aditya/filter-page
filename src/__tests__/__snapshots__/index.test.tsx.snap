// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Budget Filter Component Test > Budget Component Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span>
        <h4
          class="filter-label"
        >
          Budget
        </h4>
      </span>
      <div
        class="budget"
      >
        <div
          class="budget-selector"
        >
          <input
            id="budget-min"
            name="budget-min"
            type="number"
            value="0"
          />
          <h5
            class="filter-unit"
          >
            Lakh
          </h5>
        </div>
        <h5
          class="hyphen"
        >
          -
        </h5>
        <div
          class="budget-selector"
        >
          <input
            id="budget-max"
            name="budget-max"
            type="number"
            value=""
          />
          <h5
            class="filter-unit"
          >
            Lakh
          </h5>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <span>
      <h4
        class="filter-label"
      >
        Budget
      </h4>
    </span>
    <div
      class="budget"
    >
      <div
        class="budget-selector"
      >
        <input
          id="budget-min"
          name="budget-min"
          type="number"
          value="0"
        />
        <h5
          class="filter-unit"
        >
          Lakh
        </h5>
      </div>
      <h5
        class="hyphen"
      >
        -
      </h5>
      <div
        class="budget-selector"
      >
        <input
          id="budget-max"
          name="budget-max"
          type="number"
          value=""
        />
        <h5
          class="filter-unit"
        >
          Lakh
        </h5>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Car Card Component Test > Card Component Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="card"
      >
        <img
          alt="Car image not available"
          class="car-image"
          src="https://imgd.aeplcdn.com/0x0/cw/static/icons/svg/usedcar/no-car-image-with-text.svg"
        />
        <div
          class="car-data"
        >
          <h2
            class="car-name"
          >
            Some Car Name
          </h2>
          <span
            class="car-details"
          >
            50000
              |  
            Petrol
              |  
            Mumbai
          </span>
          <div
            class="car-pricing"
          >
            <h1
              class="car-price"
            >
              Rs. 
              5 Lakh
            </h1>
            <span
              class="car-emi"
            >
              EMI Starts at ₹4,751
            </span>
          </div>
          <button
            class="car-get-seller-details-btn"
          >
            <span>
              Get Seller Details
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="card"
    >
      <img
        alt="Car image not available"
        class="car-image"
        src="https://imgd.aeplcdn.com/0x0/cw/static/icons/svg/usedcar/no-car-image-with-text.svg"
      />
      <div
        class="car-data"
      >
        <h2
          class="car-name"
        >
          Some Car Name
        </h2>
        <span
          class="car-details"
        >
          50000
            |  
          Petrol
            |  
          Mumbai
        </span>
        <div
          class="car-pricing"
        >
          <h1
            class="car-price"
          >
            Rs. 
            5 Lakh
          </h1>
          <span
            class="car-emi"
          >
            EMI Starts at ₹4,751
          </span>
        </div>
        <button
          class="car-get-seller-details-btn"
        >
          <span>
            Get Seller Details
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Fuel Filter Component > Fuel Filter Component Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span>
        <h4
          class="filter-label"
        >
          Fuel
        </h4>
      </span>
      <div
        class="fuel"
      >
        <div
          class="fuel-selector"
        >
          <input
            id="Petrol"
            name="Petrol"
            type="checkbox"
          />
          <h5
            class="fuel-unit"
          >
            Petrol
          </h5>
        </div>
        <div
          class="fuel-selector"
        >
          <input
            id="Diesel"
            name="Diesel"
            type="checkbox"
          />
          <h5
            class="fuel-unit"
          >
            Diesel
          </h5>
        </div>
        <div
          class="fuel-selector"
        >
          <input
            id="Cng"
            name="Cng"
            type="checkbox"
          />
          <h5
            class="fuel-unit"
          >
            Cng
          </h5>
        </div>
        <div
          class="fuel-selector"
        >
          <input
            id="Lpg"
            name="Lpg"
            type="checkbox"
          />
          <h5
            class="fuel-unit"
          >
            Lpg
          </h5>
        </div>
        <div
          class="fuel-selector"
        >
          <input
            id="Electric"
            name="Electric"
            type="checkbox"
          />
          <h5
            class="fuel-unit"
          >
            Electric
          </h5>
        </div>
        <div
          class="fuel-selector"
        >
          <input
            id="Hybrid"
            name="Hybrid"
            type="checkbox"
          />
          <h5
            class="fuel-unit"
          >
            Hybrid
          </h5>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <span>
      <h4
        class="filter-label"
      >
        Fuel
      </h4>
    </span>
    <div
      class="fuel"
    >
      <div
        class="fuel-selector"
      >
        <input
          id="Petrol"
          name="Petrol"
          type="checkbox"
        />
        <h5
          class="fuel-unit"
        >
          Petrol
        </h5>
      </div>
      <div
        class="fuel-selector"
      >
        <input
          id="Diesel"
          name="Diesel"
          type="checkbox"
        />
        <h5
          class="fuel-unit"
        >
          Diesel
        </h5>
      </div>
      <div
        class="fuel-selector"
      >
        <input
          id="Cng"
          name="Cng"
          type="checkbox"
        />
        <h5
          class="fuel-unit"
        >
          Cng
        </h5>
      </div>
      <div
        class="fuel-selector"
      >
        <input
          id="Lpg"
          name="Lpg"
          type="checkbox"
        />
        <h5
          class="fuel-unit"
        >
          Lpg
        </h5>
      </div>
      <div
        class="fuel-selector"
      >
        <input
          id="Electric"
          name="Electric"
          type="checkbox"
        />
        <h5
          class="fuel-unit"
        >
          Electric
        </h5>
      </div>
      <div
        class="fuel-selector"
      >
        <input
          id="Hybrid"
          name="Hybrid"
          type="checkbox"
        />
        <h5
          class="fuel-unit"
        >
          Hybrid
        </h5>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Main Component Tests > Main Component Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="main"
      >
        <div
          class="top-filter"
        >
          <div
            class="sort-by"
          >
            <p
              class="sort-label"
            >
              Sort By:
            </p>
            <select
              class="sort-by-select"
              id="sort-by"
              name="sort-by"
            >
              <option
                value="best-match"
              >
                Best Match
              </option>
              <option
                value="price-low-to-high"
              >
                Price - Low to High
              </option>
              <option
                value="price-high-to-low"
              >
                Price - High to Low
              </option>
            </select>
          </div>
        </div>
        <div
          class="main-content"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="main"
    >
      <div
        class="top-filter"
      >
        <div
          class="sort-by"
        >
          <p
            class="sort-label"
          >
            Sort By:
          </p>
          <select
            class="sort-by-select"
            id="sort-by"
            name="sort-by"
          >
            <option
              value="best-match"
            >
              Best Match
            </option>
            <option
              value="price-low-to-high"
            >
              Price - Low to High
            </option>
            <option
              value="price-high-to-low"
            >
              Price - High to Low
            </option>
          </select>
        </div>
      </div>
      <div
        class="main-content"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
